prov/sockets/src/sock_cq.c

```diff
@@ -368,6 +368,8 @@ static ssize_t sock_cq_sreadfrom(struct fid_cq *cq, void *buf, size_t count,
 				ret = sock_cq_rbuf_read(sock_cq, buf,
 					MIN(threshold, avail / cq_entry_len),
 					src_addr, cq_entry_len);
+			else /* No CQ entry available, read the fd */
+				rbfdreset(&sock_cq->cq_rbfd);
```

Comment:
gurgle-looking at the code, I'm not sure how to handle this.  Using error flags requires re-working the socket completion handling code.  And if we go that far, we may as well replace it with the util cq code.  I would try to minimize the changes here, but avoid reading an fd every time that CQ is read.  (I don't think we end up in that situation, but I'll check the series again.)